Automattic/jetpack
# Lazy Images: Fix conflict with photon after switching to use srcset

Some JavaScript that sets height and width for Photon images had some checks for `data-lazy-src` that would postpone adding those height and width details. Here's an example:

https://github.com/Automattic/jetpack/blob/dc3e81bd139f7aeaac80513bf48e2177467f9b0a/modules/photon/photon.js#L15

After switching to use srcset instead of src for lazy loading, and removing `data-lazy-src`, the height and width attributes were being set prematurely, while the image was still 1x1. 😞 

#### Changes proposed in this Pull Request:

Add `data-lazy-src` back so that the photon image size checks are postponed until the image is actually loaded.

Ideally, we should probably make changes directly to the `photon.js` and `devicepx.js` files. But, I'd rather postpone that until we verify that `srcset` will work in the wild as expected.

#### Testing instructions:

On a connected site:

- Enable photon and lazy images
- Create a post and insert an image
- View source of post and remove any height and width attributes that were added
- Load post
- Ensure that image does load but is 1x1 😞 
- Checkout this branch
- Load post again
- Ensure image loads at proper size

#### Proposed changelog entry for your changes:
None
